bitnami/mxnet/templates/server-statefulset.yml

```diff
@@ -0,0 +1,188 @@
+        app.kubernetes.io/name: {{ include "mxnet.name" . }}
+        helm.sh/chart: {{ include "mxnet.chart" . }}
+        app.kubernetes.io/instance: {{ .Release.Name }}
+        app.kubernetes.io/component: "server"
+    spec:
+      {{- include "mxnet.imagePullSecrets" . | nindent 6 }}
+      {{- if .Values.securityContext.enabled }}
+      securityContext:
+        fsGroup: {{ .Values.securityContext.fsGroup }}
+        runAsUser: {{ .Values.securityContext.runAsUser }}
+      {{- end }}
+      {{- if .Values.nodeSelector }}
+      nodeSelector: {{ toYaml .Values.server.nodeSelector | nindent 8 }}
+      {{- end }}
+      {{- if .Values.tolerations }}
+      tolerations: {{ toYaml .Values.server.tolerations | nindent 8 }}
+      {{- end }}
+      {{- if .Values.affinity }}
+      affinity: {{ toYaml .Values.server.affinity | nindent 8 }}
+      {{- end }}
+      {{- if .Values.cloneFilesFromGit.enabled }}
+      initContainers:
+      - name: git-clone-repository
+        image: {{ include "git.image" . }}
+        imagePullPolicy: {{ .Values.git.pullPolicy | quote }}
+        command:
+        - /bin/sh
+        - -c
+        - |
+          git clone {{ .Values.cloneFilesFromGit.repository }} /app
+          cd /app
+          git checkout {{ .Values.cloneFilesFromGit.revision }}
+        volumeMounts:
+        - name: git-cloned-files
+          mountPath: /app
+      {{- end }}
+{{- if .Values.initContainers }}
+{{ toYaml .Values.initContainers | indent 6 }}
+{{- end }}
+      containers:
+      - name: mxnet
+        workingDir: {{ .Values.entrypoint.workDir }}
+        image: {{ include "mxnet.image" . }}
+        imagePullPolicy: {{ .Values.image.pullPolicy }}
+        command: 
+        - bash
+        - -c
+        - |
+          SERVER_RANK=${POD_NAME##*-}
+          ((SERVER_RANK++))
+          export SERVER_RANK
+{{ include "mxnet.entrypoint" . | indent 10 }}
+        env:
+        - name: POD_NAME
+          valueFrom:
+            fieldRef:
+              fieldPath: metadata.name
+        - name: DMLC_ROLE
+          value: "server"
+        - name: DMLC_NUM_WORKER
+          value: {{ .Values.workerCount | quote }}
+        - name: DMLC_NUM_SERVER
+          value: {{ .Values.serverCount | quote }}
+        - name: DMLC_PS_ROOT_URI
+          value: {{ include "mxnet.fullname" . }}-scheduler
+        - name: DMLC_PS_ROOT_PORT
+          value: {{ .Values.schedulerPort | quote }}
+{{- if .Values.commonExtraEnvVars }}
+{{ include "mxnet.parseEnvVars" .Values.commonExtraEnvVars | indent 8 }}
+{{- end }}
+{{- if .Values.serverExtraEnvVars }}
+{{ include "mxnet.parseEnvVars" .Values.serverExtraVars | indent 8 }}
+{{- end }}
+        {{- if .Values.livenessProbe.enabled }}
+        livenessProbe:
+          exec:
+            command:
+            - python3
+            - -c
+            - import os; os.environ["DMLC_PS_ROOT_URI"] = "127.0.0.1"; os.environ["DMLC_ROLE"] = "worker"; import mxnet; print(mxnet.__version__)
```

Comment:
The DMLC_ROLE should be 'server' no?
Same comment for the readiness

Reply:
If I do not set this value, this process will try to connect to the scheduler node, causing a conflict in the cluster

Reply:
ok ok! I thought you needed to specify the kind of node